### ISSUE TITLE
Sensors.

### ISSUE THREAD

**Issue by DaloLorn** [Alarcarr/ABEMMOD#30]
- [x] Planets have a base sight range of 1500 units.
- [x] Orbitals have a base sight range of 1000 units.
- [x] Ships without sensors have a base sight range of 250 units. Stations have a base sight range of 500.
- [x] Stuff changes an object's sight range by a flat number or a percentage, with percentile buffs/debuffs stacking either additively or multiplicatively, depending on their level (their equivalent of an EvaluationOrder). (Base sight range always comes first - 2000, 1500, 250 or whatever it is. Sensors are level 100 (negative indexes are not possible). Nebulae are level 200 percentile reductions, meaning their buffs/debuffs are applied multiplicatively to base range and levels 0 through 199, and stack additively with level 200 buffs/debuffs of any kind.)

EDIT: Okay, the framework is laid out. Now it's time to figure out what to do with that framework.
- [x] Reach a certain point where we feel satisfied with the amount of sensor content in the mod.

**Comment by DaloLorn** [comment]
No longer feasible because it requires precise and continuing knowledge of any changes to the scripts it would modify.

**Comment by Alarcarr** [comment]
Well i think that this is still feasible , i can check the sripts for u, and dont think the devs are angry with u or anything else  so there are still there for any help u need .Dont give up!

**Comment by DaloLorn** [comment]
... I removed that comment _yesterday_.

That said, this isn't a matter of 'devs are angry at me' or 'devs can't help me anymore'. It's a matter of 'don't have precise knowledge of what I need to change and how I need to change it in order to maintain compatibility with additions or fixes to the base game. (Worst-case scenario would have been that we'd have to remove all of our modified scripts in order for the mod to be capable of running.)

I've kind of worked around the issue by making a Git repository of the whole game as it's installed on my PC, but the lack of advance warning means I can only fix this stuff in response to it breaking, rather than responding to the changes as they are being made.

In other words, we're not going to have to give up _anything_, but we _are_ going to have a longer recovery period whenever the devs do something that breaks the mod.

**Comment by DaloLorn** [comment]
Okay, I just need to figure out how to sort the sight modifiers, and everything else will be a cakewalk by comparison.

So far, I have the code for all the other stuff in server/LeaderAI, it just hasn't been pushed to GitHub. Wrap up the sorting, and then I can move on to shadow/LeaderAI and data/leader_ai followed by writing up new hooks.

The first test: KHAAAAAAAAAN. (In other words, nebulae.)

**Comment by DaloLorn** [comment]
The release of 0.4.0 means that this issue is finished.
